internal/cmd/commands/targetscmd/funcs.go

```diff
@@ -676,13 +676,20 @@ func printItemTable(result api.GenericResult) string {
 	maxLength := base.MaxAttributesLength(nonAttributeMap, item.Attributes, keySubstMap)
 
 	var hostSourceMaps []map[string]interface{}
+	// used as a key to ensure there are no repeat printed values among HostSources and HostSets
+	type setInfo struct {
+		id            interface{}
+		hostCatalogID interface{}
+	}
+	hostSourceInfo := make(map[setInfo]map[string]interface{})
```

Comment:
So the need for the map in `CredentialLibraries`/`CredentialSources` makes sense because we want to group them by `Purpose`.  I am not sure its the best approach here, a slightly more maintainable and readable approach would be to use a switch statement, so something like:

```
switch {
case len(item.HostSets) > 0:
    // use HostSets
case len(item.HostSources) > 0:
   // use HostSources
}
```

I also have a hunch that the usage of the map in the credential library case was not by design to avoid duplication but rather a side effect of grouping them by purpose.

Reply:
I agree your suggestion makes more sense. I can test the implementation, but I didn't start in this direction because I had a hunch that this wouldn't work since the json data had host_sets and host_sources both populated.

Reply:
That is a great point, so the reason for this is backward compatibility.  This means we will only have one of the following scenarios:
1) Only `HostSets` is populated
2) Both `HostSets` and `HostSources` are populated, but both have the exact same value.